### PR TITLE
feat: added render prop to Sider components

### LINKED ISSUE #2453: [FEAT] Material UI and Ant Design Sider Component Modification
**Is your feature request related to a problem? Please describe.**
We had to make a custom sider to make minor changes while using both `Material UI` and `Ant Design` `Sider` component. With this feature request, we will be able to modify the top and bottom parts of the sider component by keeping the menu part the same.

**Describe the solution you'd like**
We can add two property to both `Sider` components ( topSection and bottomSection ) to reach and modify these parts. In this way we can remove or replace `Logout` button and also `Dashboard` menu item.

**Additional context**
Example of remove `Logout` button

```
<Refine
    ...
    Sider={<DefaultSider bottomSection={null} />}
    ...
/>;
```

### PR DESCRIPTION
**Pull request by ozkalai**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

`@pankod/refine-antd` `@pankod/refine-mui`

Added `render` prop to `Sider` component. You can get `dashboard`, `logout` and `items` from `render` props to customize the `Sider` component.

`@pankod/refine-ui-types`

Updated `Sider` types for `render` props.

`@pankod/refine-ui-tests`

Updated `Sider` test for `render` props.

### Test plan (required)

Screen Shot 2022-09-08 at 17 45 38
Screen Shot 2022-09-08 at 17 45 19

### Closing issues

closes #2453

### Self Check before Merge

-   [X] Corresponding issues are created/updated or not needed
-   [X] Docs are updated/provided or not needed
-   [X] Examples are updated/provided or not needed
-   [X] TypeScript definitions are updated/provided or not needed
-   [X] Tests are updated/provided or not needed
-   [X] Changesets are provided or not needed

### CODE REVIEW

**Review by aliemir — APPROVED** [review]

**Review by yildirayunlu — APPROVED** [review]
🎉

File: packages/ui-types/src/types/layout.tsx [hunk 1]
```diff
@@ -2,7 +2,14 @@ import { TitleProps, LayoutProps } from "@pankod/refine-core";
 
 export type RefineLayoutHeaderProps = {};
 
-export type RefineLayoutSiderProps = {};
+export type RefineLayoutSiderProps = {
+    bottomSection?:
+        | React.ReactNode
+        | ((defaultBottom: React.ReactNode) => React.ReactNode);
+    topSection?:
```
Comment (salihozdemir) [inline packages/ui-types/src/types/layout.tsx:9]:
```suggestion
    bottom?:
        | React.ReactNode
        | ((defaultBottom: React.ReactNode) => React.ReactNode);
    top?:
```

How is it with the new names like this?

File: packages/antd/src/components/layout/sider/index.tsx [hunk 2]
```diff
@@ -93,6 +96,45 @@ export const Sider: React.FC<RefineLayoutSiderProps> = () => {
         });
     };
 
+    const defaultTop = hasDashboard ? (
+        <Menu.Item
```
Comment (omeraplak) [inline packages/antd/src/components/layout/sider/index.tsx:100]: Could we wrap the `<CanAccess />` component?

File: packages/mui/src/components/layout/sider/index.tsx [hunk 3]
```diff
@@ -223,87 +226,104 @@ export const Sider: React.FC<RefineLayoutSiderProps> = () => {
         });
     };
 
+    const defaultTop = hasDashboard ? (
+        <Tooltip
```
Comment (omeraplak) [inline packages/mui/src/components/layout/sider/index.tsx:230]: Could we wrap the `<CanAccess />` component?

**Review by omeraplak — APPROVED** [review]

**Review by yildirayunlu — APPROVED** [review]

**Pull request merged**